src/units/abilities.cpp

```diff
@@ -839,19 +839,19 @@ std::vector<std::pair<t_string, t_string>> attack_type::special_tooltips(
  * @param[in] sp reference to ability to check
  * @param[in,out] checking_name the reference for checking if @name already added
  */
-static void add_name(std::string& weapon_abilities, bool active, const config::any_child sp, std::set<std::string>& checking_name, bool affect_adjacent)
+static void add_name(std::string& weapon_abilities, bool active, const config::any_child sp, std::set<std::string>& checking_name, bool& already_used, std::string& from_str)
 {
 	if (active) {
 		const std::string& name = sp.cfg["name"].str();
-
 		if (!name.empty() && checking_name.count(name) == 0) {
 			checking_name.insert(name);
 			if (!weapon_abilities.empty()) weapon_abilities += ", ";
-			if (affect_adjacent) {
-				weapon_abilities += sp.cfg["affect_enemies"].to_bool() ? font::span_color(font::BAD_COLOR, name) : font::span_color(font::GOOD_COLOR, name);
-			} else {
-				weapon_abilities += font::span_color(font::BUTTON_COLOR, name);
-			}
+			if(!already_used) {
```

Comment:
This looks strange. Why is the `affect_adjacent` logic removed? Maybe there's a good reason, but it just looks strange…

Reply:
with using of tooltip for indicate origin of abiliries, using colour code don't have utility.

Reply:
So basically you're saying that this change makes that code redundant.

Reply:
yes